server/src/main/java/org/opensearch/repositories/blobstore/BlobStoreFormat.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.repositories.blobstore;
+
+import org.apache.lucene.store.IndexInput;
+import org.opensearch.common.blobstore.AsyncMultiStreamBlobContainer;
+import org.opensearch.common.blobstore.BlobContainer;
+import org.opensearch.common.blobstore.stream.write.WritePriority;
+import org.opensearch.common.blobstore.transfer.RemoteTransferContainer;
+import org.opensearch.common.blobstore.transfer.stream.OffsetRangeIndexInputStream;
+import org.opensearch.common.lucene.store.ByteArrayIndexInput;
+import org.opensearch.common.xcontent.XContentFactory;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.common.bytes.BytesReference;
+import org.opensearch.core.xcontent.ToXContent;
+import org.opensearch.core.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Locale;
+
+/**
+ * Standard format that has only content for writes. Read interface does not exist as it not yet required. This format
+ * should be used for writing data from in-memory to remote store where there is no need for checksum and the client
+ * library for the remote store has inbuilt checksum capabilities while upload and download both. This format would
+ * serialise the data in Json format and store it on remote store as is. This does not support compression yet (this
+ * can be changed as required). In comparison to {@link ChecksumBlobStoreFormat}, this format does not add any additional
+ * metadata (like header and footer) to the content. Hence, this format does not depend on {@code CodecUtil} from
+ * Lucene library.
+ *
+ * @opensearch.internal
+ */
+public class BlobStoreFormat<T extends ToXContent> {
```

Comment:
I disagree with creating a "standard" BlobStoreFormat which is usually inefficient given it serialises and stores the JSON representation. Also having checksums doesn't really hurt vs adding a strong data integrity property. Lets discuss if we really need this.

Reply:
> I disagree with creating a "standard" BlobStoreFormat which is usually inefficient given it serialises and stores the JSON representation.

We can store the data in SMILE format. Since the path is not going to be hierarchal and for the sake of ease of debugging (on readability part), JSON seems like a better option. We have an index-N file in json that keeps the index metadata in plain json format.

> Also having checksums doesn't really hurt vs adding a strong data integrity property.

I removed it in considering that we have such capabilities offered by SDK libraries already. Based on my understanding, since we are reading data from in-memory (& not from disk), we can rely on SDK capabilities for handling network transport corruption. The major reason for removing the custom checksum format was that we use Lucene's classes for deriving the header and the footer (checksum). Since the concerned file is going to be used outside the OS process, it will be difficult to use the same custom encoding as that from Lucene's classes.

Reply:
I will be changing the name to reflect that this should be only used for config upload and not the standard format for file upload.

Reply:
Made the change.